feature/search/src/main/kotlin/uk/govuk/app/search/ui/SearchScreen.kt

```diff
@@ -100,7 +108,7 @@ private fun SearchScreen(
 
         when (uiState?.resultStatus) {
             ResultStatus.Success ->
-                ShowResults(uiState!!.searchResults)
+                ShowResults(uiState!!.searchResults, onClick)
```

Comment:
We should avoid using !! if possible, something like this would work:

`uiState?.let {
            when (it.resultStatus) {
                ResultStatus.Success ->
                    ShowResults(it.searchResults)
                ResultStatus.Empty ->
                    NoResultsFound(searchTerm = it.searchTerm)
                ResultStatus.DeviceOffline ->
                    DeviceIsOffline()
                ResultStatus.ServiceNotResponding ->
                    ServiceNotResponding()
                else ->
                    ShowNothing()
            }
        } ?: ShowNothing()`

Reply:
Fixed up